src-self-hosted/astgen.zig

```diff
@@ -121,6 +123,43 @@ fn assign(mod: *Module, scope: *Scope, infix_node: *ast.Node.SimpleInfixOp) Inne
     }
 }
 
+/// Identifier nodes -> ZIR string instruction
+pub fn identifierString(mod: *Module, scope: *Scope, node: *ast.Node.Identifier) InnerError!*zir.Inst {
+    const tree = scope.tree();
+    const src = tree.token_locs[node.token].start;
+
+    var ident_name = tree.tokenSlice(node.token);
+    if (std.mem.startsWith(u8, ident_name, "@"))
+        ident_name = ident_name[2 .. ident_name.len - 1];
```

Comment:
We'll need a more robust way to handle `@"aoeu"` tokens. Consider it could be `@"hello\n"` in which case this would get the wrong identifier name. When it is a special token like this zig will have to allocate memory to parse the string, using `std.zig.parseStringLiteral`. The good news is that there is already an arena to use for this allocation and you can use `scope.arena()` to get it.

There are a few more instances of this (written by me, marked by "TODO" comments) around. I suggest to either solve all of them in this patch or to mark this instance in the same pattern as the others, and tackle it in a future patch.